### PR TITLE
Fix link error for LLD 12.0.3

### PR DESCRIPTION
**Pull request by James-Zzz**

### CODE REVIEW

File: lib/efi_services/efi_ap.S [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 
 .extern cpu_num
-.extern hand_over_entry
+.extern efi_hand_over_entry
```
Comment (kwang13) [inline lib/efi_services/efi_ap.S:10]: what's the purpose for this change?

Comment (James-Zzz) [reply lib/efi_services/efi_ap.S:10]: There is also a global symbol hand_over_entry exists in lib/mp_init/mpinit.c.
So rename hand_over_entry in lib/efi_services to address duplicated symbol.

**Review by kwang13 — APPROVED** [review]
